### PR TITLE
DWR-652 - Persist report contents using the common ArchiveService

### PR DESCRIPTION
**Pull request by wtritch**

### CODE REVIEW

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveBackedReportContentService.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import java.io.InputStream;
+import java.net.URI;
+
+/**
+ * This implementation of a ReportContentService is backed by an ArchiveService.
+ */
+@Service
+public class ArchiveBackedReportContentService implements ReportContentService {
+    private final static String BasePath = "REPORTS/";
```
Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveBackedReportContentService.java]: This wraps the ArchiveService with Report knowledge.  I'm going to stuff the reports into a configurable bucket under a "/REPORTS/" directory.

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportService.java [hunk 2]
```diff
@@ -34,11 +33,6 @@ public Report insert(final Report report) {
     }
 
     @Override
-    public InputStream getContent(final long reportId) {
-        return null;
-    }
```
Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/DefaultReportService.java]: I pulled this out into a separate ReportContentService.

**Comment by wtritch** [comment]
Merging in the name of time, will continue to resolve comments as they come in.

File: pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveBackedReportContentService.java (hunk 1 above)
Comment (malaffoon) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveBackedReportContentService.java]: There are going to be a lot of `REPORTS`, perhaps we should be more specific; or add a subfolder. For example, we will have import reconciliation reports, aggregate reports, reliability reports. So, not sure how to decorate this one, maybe start with `REPORTS/USER/` or `USER_REPORTS` or ???

Comment (wtritch) [inline pdf-report-processor/src/main/java/org/opentestsystem/rdw/reporting/processor/service/ArchiveBackedReportContentService.java]: I like that idea.  REPORTS/USER/ it is

**Pull request merged**